test/test-tty.c

```diff
@@ -28,7 +28,7 @@
 #else /*  Unix */
 # include <fcntl.h>
 # include <unistd.h>
-# if defined(__linux__)
+# if defined(__linux__) || (defined(__FreeBSD_kernel__) && defined(__GNUC__))
```

Comment:
Hi,

GNU libc provides pty.h, and the correct macro to detect that is __GLIBC__  (whereas __GNUC__ refers to the GCC compiler being used).

But also, I think it should be sufficient to say:

# if defined(__linux__) || defined(__GLIBC__)

since GNU libc is expected to provide pty.h also on GNU/Hurd, or GNU/anything else that may (someday) exist.  This is not something specific to GNU/kFreeBSD or relating to the kernel.